src/sidebar/util/test/service-context-test.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const propTypes = require('prop-types');
+const { createElement, render } = require('preact');
+
+const { ServiceContext, withServices } = require('../service-context');
+
+describe('service-context', () => {
+  describe('withServices', () => {
+    let container;
+    let lastProps;
+
+    function TestComponent(props) {
+      lastProps = props;
+    }
+    TestComponent.injectedProps = ['aService'];
+    const WrappedComponent = withServices(TestComponent);
+
+    beforeEach(() => {
+      lastProps = null;
+      container = document.createElement('div');
+    });
+
+    it('returns the input component if there are no service dependencies', () => {
+      function TestComponent() {}
+      assert.equal(withServices(TestComponent), TestComponent);
+    });
+
+    it('looks up services that a Component depends on and injects them as props', () => {
+      const testService = {};
+      const injector = {
+        get: sinon.stub().returns(testService),
+      };
+      render(
+        <ServiceContext.Provider value={injector}>
+          <WrappedComponent />
+        </ServiceContext.Provider>,
+        container
+      );
+      assert.deepEqual(lastProps, { aService: testService });
+      assert.calledWith(injector.get, 'aService');
+    });
+
+    it('copies propTypes except for injected properties to wrapper', () => {
+      function TestComponent() {}
+      TestComponent.propTypes = {
+        notInjected: propTypes.string,
+        injected: propTypes.string,
+      };
+      TestComponent.injectedProps = ['injected'];
+
+      const Wrapped = withServices(TestComponent);
+
+      assert.deepEqual(Wrapped.propTypes, { notInjected: propTypes.string });
+      assert.isUndefined(Wrapped.injectedProps);
```

Comment:
I see where individual properties are removed from `injectedProps` but not where the Object is deleted in entirety (i.e. I don't understand why the object is undefined here).

Reply:
When the `Wrapper` component is created in `withServices`, it has no `injectedProps` property in the first place.